src/raw/mod.rs

```diff
@@ -29,6 +35,23 @@ impl<K, V> Table<K, V> {
         self.bins.len()
     }
 
+    pub(crate) fn get_moved<'g>(
+        &'g self,
+        for_table: *const Table<K, V>,
+        guard: &'g Guard,
+    ) -> Shared<'g, BinEntry<K, V>> {
+        match self.moved.load(Ordering::SeqCst, guard) {
+            s if s.is_null() => {
+                // if a BinEntry::Moved associated with this table does not yet exist,
+                // create one and store it in `self.moved`
+                self.moved
+                    .store(Owned::new(BinEntry::Moved(for_table)), Ordering::SeqCst);
+                self.moved.load(Ordering::SeqCst, guard)
+            }
+            s => s,
```

Comment:
We should probably `assert!` here that `for_table` matches what is in `s`.

Reply:
Good point!